package com.myproject.spring5.util.sftp;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import com.jcraft.jsch.JSchException;
import com.jcraft.jsch.SftpException;

@RunWith(SpringJUnit4ClassRunner.class)
@ContextConfiguration(classes = SftpBeanConfig.class)
public class SFtpServiceTest {

	static final Logger logger = LogManager.getLogger(SFtpServiceTest.class);
	
	@Autowired
	private SFtpService sFtpService;
	
	@Test
	public void testFtpConnetion(){

		SFtpContext sftpContext = new SFtpContext();
		sftpContext.setServerIP("10.46.54.14");
		sftpContext.setPort(22);
		sftpContext.setUserId("wasadmin");
		sftpContext.setPassword("!adt0305");
		
		try {
			try {
				sFtpService.setContext(sftpContext);
			} catch (SftpException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
		} catch (JSchException e) {
			logger.error("업로드에 실패하였습니다. 원인 : {}",e.getMessage());
		}
	}
}
